[Nevin Lim] iP

nevinlim opened the pull request:
# Duke bot
>Simplicity is the soul of efficiency. -Austin Freeman [[ source](http://www.vanderburg.org/Misc/Quotes/soft-quotes.html)]

Duke frees your mind of having to remember things you need to do. It's:

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nevinlim/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks. 
4. let it manage your tasks for you 😉

And it is **FREE**❗ ❗ 😃 

Features:
- [ ]  Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

`````ruby
public static void main(String[] args) {
        System.out.println("Hello! I'm Duke");
    }
`````

Commit:
Add Gradle support

seanleong339 committed:
Level-1

nevinlim committed:
Revert "Level-1"

This reverts commit ba7321493f7778eef668eb22592632e937f67a7b.

nevinlim committed:
Level-1

nevinlim committed:
Level-2

nevinlim committed:
Level-3

nevinlim committed:
Level-4

nevinlim committed:
Level-5

nevinlim committed:
A-TextUiTesting

nevinlim committed:
A-TextUiTesting

nevinlim committed:
A-TextUiTesting

nevinlim committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Emrysil commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
Would it be better to use private modifier here? Or is there a particular reason to use protected?

Emrysil commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,123 @@
+                    System.out.println("____________________________________________________________\n" +
+                            "Nice! I've marked this task as done:\n" +
+                            "   " + tasks.get(num - 1).toString() + "\n" +
+                            "____________________________________________________________");
```
I think you could make the line breaks private fields so that you don't need to type so many times

Emrysil commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,123 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static final String greeting = "____________________________________________________________\n" + "Hello! I'm Duke\n" + "What can I do for you?\n" + "____________________________________________________________";
```
I think for static final fields, the variable name should all be in uppercase:)

Emrysil commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,123 @@
 public class Duke {
+    private static final String greeting = "____________________________________________________________\n" + "Hello! I'm Duke\n" + "What can I do for you?\n" + "____________________________________________________________";
+    private static final String goodbye = "____________________________________________________________\n" + "Bye. Hope to see you again soon!\n" + "____________________________________________________________";
+    private static boolean active = true;
```
perhaps rename to isActive to make the variable sound more boolean?

Emrysil commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+import java.util.*;
```
Maybe specify the class being imported instead of just importing *

nevinlim committed:
Level-7

nevinlim committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

nevinlim committed:
Level-7

nevinlim committed:
Level-8

nevinlim committed:
Merge branch 'branch-level-8'

nevinlim committed:
A-MoreOOP

nevinlim committed:
A-Packages

nevinlim committed:
A-JUnit

nevinlim committed:
A-Jar

nevinlim committed:
A-JavaDoc

nevinlim committed:
A-CodingStandard

nevinlim committed:
Level-9

nevinlim committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/main/Parser.java

Tariq-droid commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,22 @@
+
+import java.io.IOException;
+
+public class MarkCommand extends Command {
```
need javadoc

Tariq-droid replied:
I don't think it makes sense to have these as classes. methods make more sense

Tariq-droid commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    public void execute(Tasklist tasklist, Ui ui, Storage storage) throws IOException {
+        tasklist.markDone(this.taskNum - 1);
```
tasklist -> taskList

Tariq-droid commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,111 @@
+import duke.command.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
```
should split the imports

Tariq-droid commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,111 @@
+     * @throws DukeException when input is invalid
+     */
+    public static Command parseCommand(String fullCommand) throws DukeException {
+        String[] splits = fullCommand.split(" ", 2 );
```
not a very indicative name for this array

Tariq-droid commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,111 @@
+                throw new DukeException("☹ OOPS!!! Please use format: event <description> /from <datetime> /to <datetime>");
+            }
+            try {
+                String[] secondsplits = splits[1].split("/from", 2);
```
variable naming standards

Tariq-droid commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,129 @@
+        StringBuilder strings = new StringBuilder();
+        FileWriter fw = new FileWriter(this.file);
+
+        for (Task curr : tasklist.getTasks()) {
```
this  is a complicated expression. Can be simplified

Tariq-droid commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,129 @@
+                        .append(curr.getDescription())
+                        .append("\n");
+            } else if (curr instanceof Deadline) {
+                strings.append("D ")
```
not very readable

Tariq-droid commented on src/main/java/duke/main/Ui.java:
```diff
@@ -0,0 +1,156 @@
+        printLine();
+    }
+
+    public void printFoundTasks(ArrayList<Task> foundTasks) throws DukeException {
```
missing javadoc

Tariq-droid commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,11 @@
+import duke.main.Ui;
+
+public class ExitCommand extends Command {
+    public void execute(Tasklist taskList, Ui ui, Storage storage) {
```
nota very indicative name for the method

nevinlim committed:
Level-10

nevinlim committed:
A-Assertions

nevinlim committed:
Merge pull request #1 from nevinlim/A-Assertions

A-Assertions

nevinlim committed:
A-CodeQuality

nevinlim committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java

nevinlim committed:
Tasklist: Removed duplicate method

Removed duplicated getTasksNum() method from Tasklist and did some cosmetic changes to several other classes.

nevinlim committed:
Update README.md according to GFMD standards

nevinlim committed:
Updated README.md again

nevinlim committed:
C-DetectDuplicates

Added the ability to recognize and deal with duplicate tasks from user.

nevinlim committed:
Merge remote-tracking branch 'origin/master'

nevinlim committed:
A-Personality

Added *personality* to newly named Smudge program. Added new help command too

nevinlim committed:
User Guide

nevinlim committed:
UI screenshot

nevinlim committed:
Update README.md

nevinlim committed:
Update README.md

nevinlim committed:
updated User Guide

nevinlim committed:
updated UG

nevinlim committed:
updated UG

nevinlim committed:
Update README.md

nevinlim committed:
updated UG

nevinlim committed:
update UG
